cni-plugin/linkerd-cni-validator/src/main.rs

```diff
@@ -0,0 +1,256 @@
+use std::{
+    net::{IpAddr, SocketAddr},
+    process::exit,
+    sync::Arc,
+    time,
+};
+
+use anyhow::{anyhow, Context, Result};
+use clap::Parser;
+use rand::distributions::{Alphanumeric, DistString};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{TcpListener, TcpStream},
+    sync::Notify,
+};
+use tracing::{debug, error, info, Instrument};
+
+#[derive(Parser)]
+#[clap(version)]
+struct Args {
+    #[clap(
+        long,
+        env = "LINKERD_VALIDATION_LOG_LEVEL",
+        default_value = "linkerd_cni_validator=info,warn"
+    )]
+    log_level: kubert::LogFilter,
+
+    #[clap(long, default_value = "plain")]
+    log_format: kubert::LogFormat,
+
+    #[clap(long, default_value = "0.0.0.0:4140")]
+    outbound_proxy_addr: SocketAddr,
+
+    #[clap(parse(try_from_str), long, env = "KUBERNETES_SERVICE_HOST")]
+    target_ip: IpAddr,
+
+    #[clap(parse(try_from_str), long, env = "KUBERNETES_SERVICE_PORT")]
+    target_port: u16,
+
+    #[clap(parse(try_from_str = parse_timeout), long, default_value = "120s")]
+    timeout: std::time::Duration,
+}
+
+// ERRNO 95: Operation not supported
+const UNSUCCESSFUL_EXIT_CODE: i32 = 95;
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() -> Result<()> {
+    let Args {
+        log_level,
+        log_format,
+        outbound_proxy_addr,
+        target_ip,
+        target_port,
+        timeout,
+    } = Args::parse();
+
+    log_format.try_init(log_level)?;
+    let target_addr = SocketAddr::new(target_ip, target_port);
+
+    let rng_resp = Alphanumeric.sample_string(&mut rand::thread_rng(), 8);
+    info!(%outbound_proxy_addr, %target_addr, "Validating outbound traffic is redirected to proxy's outbound port");
+
+    let (shutdown_tx, shutdown_rx) = kubert::shutdown::sigint_or_sigterm()?;
+    let notify_ready = Arc::new(Notify::new());
+
+    let server_task = tokio::spawn(outbound_serve(
+        outbound_proxy_addr,
+        notify_ready.clone(),
+        shutdown_rx,
+        rng_resp.clone(),
+    ));
+
+    let validation_task = tokio::spawn(validate_outbound_redirect(
+        target_addr,
+        notify_ready,
+        timeout,
+        rng_resp,
+    ));
+
+    tokio::select! {
+        task = validation_task => {
+            if let Err(err) = task.expect("Failed to run validator task") {
+                error!(error = %err, "Failed validation");
+                exit(UNSUCCESSFUL_EXIT_CODE);
+            }
+
+            info!("Validation passed successfully...exiting");
+            Ok(())
+        }
+
+        _ = shutdown_tx.signaled() => {
+            if let Err(e) = server_task.await.expect("Failed to run outbound server task") {
+                error!(error = %e, "Failed to validate outbound routing configuration");
```

Comment:
similarly, maybe just write
```suggestion
            if let Err(error) = server_task.await.expect("Failed to run outbound server task") {
                error!(%error, "Failed to validate outbound routing configuration");
```